frontend/staff-dashboard/src/components/flexiblepricing/statusmodal.tsx

```diff
@@ -43,6 +43,17 @@ export const FlexiblePricingStatusModal: React.FC<IFlexiblePriceStatusModalProps
     const { mutate } = mutationResult;
 
     const [ justification, setJustification ] = useState(modaldata.justification);
+    const [ discount, setDiscount ] = useState(modaldata.discount);
+    let discount_choices = [];
+    for (let applicable_discount of modaldata.applicable_discounts) {
+        discount_choices.push(
+            {
+                "value": applicable_discount.id,
+                "label": applicable_discount.amount + " " + applicable_discount.discount_type + " " + applicable_discount.discount_code
```

Comment:
Can we do a bit more formatting here? It'd be nice if things like extraneous decimal points were removed and the discount type were a bit more user-friendly. (E.g., in testing, my tier has a 50% discount and that's displayed as "50.00000 percent-off". It'd be nicer if it said "50% off" or "$200 off" or "$399 fixed price" or something along those lines.) Same formatting should apply in `list.tsx` too.

In addition, this is repeated in the list view - can the formatting be moved to a function so this can be DRYed up?

Reply:
Sure, On it.